Basic_viewer/include/CGAL/Qt/Basic_viewer.h

```diff
@@ -61,7 +61,7 @@
 namespace CGAL {
 
 //------------------------------------------------------------------------------
-class Basic_viewer_qt : public CGAL::QGLViewer
+class Basic_viewer : public CGAL::QGLViewer
```

Comment:
I understand that one wants to have `CGAL::Basic_viewer`. However, I know that Guillaume plans to implement other variations of that basic viewer using other graphical toolkits ([glfw](https://www.glfw.org/) was mentioned). In the future, if one wants to keep compatibility, another name will have to be used. In particular for users with a Qt application incorporating the viewer as a widget.

That is why I propose that `Basic_viewer` is in the namespace `CGAL::Qt`. Then in the `CGAL` namespace a using declaration will allow to refer to it as `CGAL::Basic_viewer`:
```c++
namespace CGAL {
  using Qt::Basic_viewer;
}
```

Reply:
Would this reasoning not also apply to the generic `draw()` function?

Reply:
> Would this reasoning not also apply to the generic `draw()` function?

I do not think so. When the user calls a `CGAL:::draw` function, a window pops up to display the graphics scene, but the user do not really care if the window is a Qt one or something else. But, in a example like that one:
https://github.com/CGAL/cgal/blob/5f9084bf67198bae2c52f22b3bd437273a250f28/Basic_viewer/examples/Basic_viewer/draw_several_windows.cpp#L68-L80
if `CGAL::Basic_viewer` is not a Qt widget, that will break the code.

Actually, I do not understand understand why you want to have `CGAL::Basic_viewer` without Qt somewhere in its name. The only other example using it is:
https://github.com/CGAL/cgal/blob/5f9084bf67198bae2c52f22b3bd437273a250f28/Basic_viewer/examples/Basic_viewer/draw_surface_mesh_small_faces.cpp#L105-L113
and that example also needs to use Qt specifics to interact with the viewer class.

Andreas, why do you want to remove the name "Qt" from that class (`Basic_viewer`). I do not think one might want to use it without using Qt specific API.

Reply:
No problem to keep it. As suffix or as namespace. The latter seems more elegant.

Reply:
namespace.

Reply:
done